src/generator/AutoRest.Go/Model/CompositeTypeGo.cs

```diff
@@ -0,0 +1,217 @@
+                if (!properties.Any(p => p.Name.EqualsIgnoreCase(nextLinkField)))
+                {
+                    var property = new PropertyGo();
+                    property.Name = nextLinkField;
+                    property.ModelType = new PrimaryTypeGo(KnownPrimaryType.String);
+                    properties.Add(property);
+                }
+            }
+
+            // Emit each property, except for named Enumerated types, as a pointer to the type
+            foreach (var property in properties)
+            {
+                var enumType = property.ModelType as EnumTypeGo;
+                if (enumType != null && enumType.IsNamed())
+                {
+                    indented.AppendFormat("{0} {1} {2}\n",
+                                    property.Name,
+                                    enumType.Name,
+                                    property.JsonTag());
+
+                }
+                else if (property.ModelType is DictionaryType)
+                {
+                    indented.AppendFormat("{0} *{1} {2}\n", property.Name, (property.ModelType as DictionaryTypeGo).Name, property.JsonTag());
+                }
+                else if (property.ModelType.PrimaryType(KnownPrimaryType.Object))
+                {
+                    // TODO: I don't think this is the best way to handle object types
+                    indented.AppendFormat("{0} *{1} {2}\n", property.Name, property.ModelType.Name, property.JsonTag());
+                }
+                else if (property.ShouldBeFlattened())
+                {
+                    // embed as an anonymous struct.  note that the ordering of this clause is
+                    // important, i.e. we don't want to flatten primary types like dictionaries.
+                    indented.AppendFormat("*{0} {1}\n", property.ModelType.Name, property.JsonTag());
+                    property.Extensions[SwaggerExtensions.FlattenOriginalTypeName] = Name;
+                }
+                else
+                {
+                    indented.AppendFormat("{0} *{1} {2}\n", property.Name, property.ModelType.Name, property.JsonTag());
+                }
+            }
+
+            return indented.ToString();
+        }
+
+        public bool IsWrapperType { get { return _wrapper; } }
+
+        public IModelType BaseType { get; set; }
+
+        public IModelType GetElementType(IModelType type)
+        {
+            if (type is SequenceTypeGo)
+            {
+                Name += "List";
+                return GetElementType((type as SequenceType).ElementType);
+            }
+            else if (type is DictionaryTypeGo)
+            {
+                Name += "Set";
+                return GetElementType(((type as DictionaryTypeGo).ValueType));
+            }
+            else
+            {
+                return type;
+            }
+        }
+
+        public string PreparerMethodName()
+        {
+            return Name + "Preparer";
+        }
```

Comment:
Methods like these should be properties, not methods, and can be specified as :

public string PreparerMethod => $"{Name}Preparer";